### ISSUE TITLE
CMS fails to load due to a bug in react-aria-menubutton

### ISSUE THREAD

**Issue by yoavniran** [decaporg/decap-cms#3603]
**Describe the bug**
CMS fails to load due to a bug in https://github.com/davidtheclark/react-aria-menubutton/

I havent seen this reported yet and im not sure how this isnt then a problem for more people because from the looks of it, it should be critical bug for all users that use netlify-cms ever since react-aria-menubutton changed their code.

The error itself is:

```

ManagerContext.js:5 Uncaught TypeError: createReactContext is not a function
    at Object../node_modules/react-aria-menubutton/dist/ManagerContext.js (ManagerContext.js:5)
    at __webpack_require__ (bootstrap:725)
    at fn (bootstrap:100)
    at Object../node_modules/react-aria-menubutton/dist/Wrapper.js (Wrapper.js:12)
    at __webpack_require__ (bootstrap:725)
    at fn (bootstrap:100)
    at Object../node_modules/react-aria-menubutton/dist/index.js (index.js:6)
    at __webpack_require__ (bootstrap:725)
    at fn (bootstrap:100)
    at Object../node_modules/netlify-cms-ui-default/dist/esm/Dropdown.js (Dropdown.js:22)

```

Its happening because in their transpiled code inside ManagerContext.js 

node_modules/react-aria-menubutton/dist/ManagerContext.js
is doing:
```javascript
var createReactContext = require('create-react-context');

var AriaMenuButtonManagerContext = createReactContext();
```

instead of: 
```javascript
var createReactContext = require('create-react-context').default;
```

If I change the code in my node_modules folder to the latter, everything goes back to work. But obviously this isnt a solution as i can only do it locally. I publish my site through netlify and there I have the bug which means users cannot update or create content anymore.

The netlify-cms package that's using this dep is: netlify-cms-ui-default

I realize this is a bug in the 3rd party library but i havent seen anyone reporting this issue on their GH repo (Ive opened a bug there as well)

In any case, the issue is that its completely broken my use of netlify-cms so Im hoping to find a solution here rather than hope they fix it on their end.

**To Reproduce**

see versions of dependencies below. As far as i know, there's nothing special about my setup that causes this problem...

**Screenshots**

<img width="2910" alt="Screen Shot 2020-04-14 at 13 20 39" src="https://user-images.githubusercontent.com/1102278/79214138-f6b9a100-7e52-11ea-9133-0f9e801cbb80.png">

**Applicable Versions:**
 - Netlify CMS version: 2.12.7
 - netlify-cms-ui-default: 2.9.5

**Additional context**
bug opened on react-aria-menubutton:  https://github.com/davidtheclark/react-aria-menubutton/issues/127

**Comment by erezrokah** [comment]
Hi @yoavniran, I've seen this recently when someone was trying to create a custom widget based on the number widget and got hit with this error when trying to use the preview component (the solution was to remove the import for the preview component). Can you elaborate on how you're using the CMS as a dependency? Possibly share an example repo?

**Comment by yoavniran** [comment]
thanks for the quick reply @erezrokah I will have a look and see if i can disable stuff and if that will have an effect.

however, if you look at the stracktrace:

```
ManagerContext.js:5 Uncaught TypeError: createReactContext is not a function
    at Object../node_modules/react-aria-menubutton/dist/ManagerContext.js (ManagerContext.js:5)
    at __webpack_require__ (bootstrap:725)
    at fn (bootstrap:100)
    at Object../node_modules/react-aria-menubutton/dist/Wrapper.js (Wrapper.js:12)
    at __webpack_require__ (bootstrap:725)
    at fn (bootstrap:100)
    at Object../node_modules/react-aria-menubutton/dist/index.js (index.js:6)
    at __webpack_require__ (bootstrap:725)
    at fn (bootstrap:100)
    at Object../node_modules/netlify-cms-ui-default/dist/esm/Dropdown.js (Dropdown.js:22)

```

the error is coming from: netlify-cms-ui-default/dist/esm/Dropdown.js (Dropdown.js:22)

so it doesnt look like it has to do with any customization of mine.

Its a little hard to share my configuration because my site is in a private repo and im using javascript to generate the CMS config (YAML is a disaster, to put it mildly, so I refuse to use it when possible) 

I am using a Youtube Editor component based on the example shown in the docs. 
I also have a couple of custom widgets I created (RTLMarkdownWidget & PostIdWidget). I will see if something changes when I disable these.

thanks

**Comment by yoavniran** [comment]
hey @erezrokah 
I removed all of my custom configuration (all collections and widgets) but still the error persists. 
As I mentioned before, the stacktrace shows that its coming from internal netlify-cms code. 
Im still puzzled how this isnt a more general issue for all users and how it wasnt reported by more people.

Not sure what im missing here.

From the  bug i opened for react-aria-menubutton I got a response that soon they will release a major version without create-react-context which will fix the problem. 

Unfortunately, until that happens and you guys upgrade the dep, Im stuck without a working CMS. I guess I'll try to downgrade until I find a working version in the meantime.

**Comment by erezrokah** [comment]
I think it will only show up if you bundle the CMS in a specific way that triggers the import. For example our demo site https://cms-demo.netlify.com/#/collections/posts uses the CMS from CDN and doesn't have that issue.

**Comment by yoavniran** [comment]
well... ive installed it through npm... Isnt that the normal way? Is there a preferable way to consume netlify-cms?
in any case, the error happens whether i build in dev or in prod...

**Comment by erezrokah** [comment]
If you could share a reproduction repo it might help. You could either use the CMS from CDN or install from npm via `netlify-cms-app`. The latter is the preferred way since it is easier to register custom code.

**Comment by yoavniran** [comment]
cool. ill try and make time to create a public repo to reproduce. thanks

**Comment by lookea** [comment]
this is breaking my site, without doing any changes on libraries just on rebuilding this issue was introduced

**Comment by erezrokah** [comment]
Hi @lookea, any chance you can setup a reproduction repo? I saw this issue when using trying to extend an existing CMS widget. Can you share your use case?

**Comment by lookea** [comment]
Hi I have workaround this issue modifying my widgets they were importing 
import { ObjectWidgetTopBar, lengths } from 'netlify-cms-ui-default';

I hope the issue get fixed in the next release of that library

**Comment by yoavniran** [comment]
thanks @lookea. unfortunately, i don have an reference to the netlify-cms-ui-default lib but still have this problem...

**Comment by yoavniran** [comment]
I've made a little progress in diagnosing the problem - 
What I noticed is that create-react-context is being overriden by Gatsby's cache. 
So its not the actual dependency (from node_modules/create-react-context) but a ES version that only exposes React's createContext method.

.cache/create-react-context.js

```javascript
import React from "react"

export default React.createContext
```
 
This explains why the require from react-aria-menubutton is breaking.

I dont know where this is coming from and if its possible to stop Gatsby from doing so. A google search I did didn't find anything useful. I will open an issue for them as well

**Comment by yoavniran** [comment]
opened bug at Gatsby repo: https://github.com/gatsbyjs/gatsby/issues/23645

**Comment by martinjagodic** [comment]
Closing as fixed
